digitalfabrik/entitlementcard
# 1084: Multi cards

### Short description

As a user i want to be able to have multiple cards f.e. for my family on one device

### Proposed changes



- added carousel slider for more than one card
- added possibility to remove single card
- import existing card from former version
- added card limit to build config

Note: i slightly changed the order of the "More Actions" and the slider indicator to avoid unneeded refactoring. It was approved by design team.

Additionally i've made the indicator fixed at the bottom and the rest scrollable

### Side effects



- hopefully none

### Resolved issues



Fixes: #1084 

Tested on ios simulator & real device and android emulator & real device

### Testing
- add cards with different barcodes and delete them
- check compatibility with old version: deploy an old app version (f.e. main branch) on the device. Activate a card. Switch to this branch and deploy. Check if the activated card still exists.
- check if you can add cards if card limit is reached
- test invalidation f.e. disable timezone on device and set a custom time, so the time is not synced with the server

Note: If you want to test it locally on ios, you can add base64 string  like in #1144 . If you want to test on ios real device you have to change `localhost` to `<your-network-ip>`
